src/hooks/useBlockList.tsx

```diff
@@ -0,0 +1,71 @@
+import { useMyAxios } from 'hooks/useMyAxios'
+import { useCallback, useState } from 'react'
+import { BlockedList, CommandBase } from 'types/custom_types'
+
+export const useBlockList = () => {
+  const [blockList, setList] = useState<BlockedList>({
+    command_publishing_blocklist: [],
+  })
+  const { axiosInstance } = useMyAxios()
+
+  const getList = useCallback(async () => {
+    const { data } = await axiosInstance.get<BlockedList>(
+      '/api/v1/commandpublishingblocklist',
+    )
+    setList(data)
+    return data
+  }, [setList, axiosInstance])
+
+  const getBlockList = () => {
```

Comment:
If you come to the page with an empty blocklist, this goes into an infinite loop.